libGitWrap/Base.hpp

```diff
@@ -42,31 +39,82 @@ namespace Git
         typedef Internal::BasePrivate Private;
 
     public:
-        Base();
-        Base(const Base& other);
-        virtual ~Base();
-        Base& operator=(const Base& other);
-        bool operator==(const Base& other) const;
-        bool operator!=(const Base& other) const;
+        Base(Internal::BasePrivate* bp = GW_NULLPTR)
```

Comment:
This does part of the trick (avoids creation of lots of temporary `Ptr` objects just to increment the refcount). Thanks - :+1:. I actually thought the same, while I was playing around with the diff stuff and we discussed the two implicit c'tors - but chose the wrong one - that's life :).